torch/testing/_internal/common_device_type.py

```diff
@@ -532,12 +531,41 @@ def setUpClass(cls):
 
 class MPSTestBase(DeviceTypeTestBase):
     device_type = 'mps'
+    primary_device: ClassVar[str]
+
+    @classmethod
+    def get_primary_device(cls):
+        return cls.primary_device
+
+    @classmethod
+    def get_all_devices(cls):
+        # currently only one device is supported on MPS backend
+        prim_device = cls.get_primary_device()
+        return [prim_device]
+
+    @classmethod
+    def setUpClass(cls):
+        cls.primary_device = 'mps:0'
+
+    @classmethod
+    def _get_dtypes(cls, test):
+        requested_dtypes : List[torch.dtype] = list()
+        # get a list of types provided by @dtypes or @dtypesIfMPS
+        test_dtypes = super()._get_dtypes(test)
+        if test_dtypes is not None:
+            requested_dtypes = list(test_dtypes)
+        if not requested_dtypes:
+            return None
+
+        from torch.testing._internal.common_dtype import integral_types_and
+        # populate list of datatypes supported on MPS device
+        supported_mps_dtypes = set(integral_types_and(torch.bool, torch.float16, torch.float32))
```

Comment:
I would prefer that we have an explicit list of non-supported devices. As it is written here, it is non-trivial to know what is ignored.

Reply:
Ok sounds good.